lib/inc/facter/ruby/ruby.hpp

```diff
@@ -41,6 +42,16 @@ namespace facter { namespace ruby {
     LIBFACTER_EXPORT void load_custom_facts(facter::facts::collection& facts, std::vector<std::string> const& paths = {});
 
     /**
+     * Traverses a ruby fact and returns a new value based on the
+     * query segments passed in the range.
+     * @param value The original value to query
+     * @param segment The beginning of the query segment range
+     * @param end The end of the query segment range
+     * @return Returns a pointer to the value queried, or nullptr if it does not exist.
+     */
+    LIBFACTER_EXPORT facts::value const* lookup(facts::value const* value, std::vector<std::string>::iterator segment, std::vector<std::string>::iterator end);
```

Comment:
I'm happy to expose ruby_value (the rest of the value types are public API, if you think that's a reasonable answer... but it probably would pull in more of the leatherman ruby stuff as public API, which I think we didn't want.